Tests/XcodeProjTests/Objects/Files/PBXGroupTests.swift

```diff
@@ -199,6 +199,77 @@ final class PBXGroupTests: XCTestCase {
         XCTAssertEqual(file!.sourceTree, .developerDir)
     }
 
+    func test_whenSearchingForFilesByPath_thenNormalizedAbsolutePathsAreUsedInternallyForUniqueness() throws {
+        let project = makeEmptyPBXProj()
+        let subDirectoryName = UUID().uuidString
+
+        func prepareDirectories() throws -> (projectDir: Path, groupDir: Path, group: PBXGroup) {
+            let absolutePath = try Path.uniqueTemporary()
+            let splitSize = 2
+            let projectDir = Path(components: absolutePath.components[0..<absolutePath.components.count-splitSize])
+            let groupDir = Path(components: absolutePath.components.reversed()[0..<splitSize].reversed())
+            XCTAssertEqual(projectDir + groupDir, absolutePath)
+
+            let subDir = absolutePath + subDirectoryName
+            try subDir.mkdir()
+
+            let group = PBXGroup(children: [], sourceTree: .group, name: "group", path: ".")
+            let parent = PBXGroup(children: [group], sourceTree: .absolute, name: "parent", path: projectDir.string)
+            project.add(object: parent)
+            project.add(object: group)
+
+            return (projectDir, groupDir, group)
+        }
+
+        let (projectDir, groupDir, theGroup) = try prepareDirectories()
+
+        let fileName = UUID().uuidString
+        let file1Path = groupDir + fileName
+        let file2Path = groupDir + subDirectoryName + fileName
+
+        let fileReferenceFor = try [file1Path, file2Path]
+            .reduce([Path: PBXFileReference]()) { map, filePath in
+                let absolutePath = projectDir + filePath
+                try Data().write(to: absolutePath.url)
+                let file = try theGroup.addFile(
+                    at: absolutePath,
+                    sourceTree: .group,
+                    sourceRoot: projectDir)
+
+                return map.merging([filePath: file]) { _, new in new }
+            }
+
+        func assert(filePath: Path, hasReference: PBXFileReference?, line: UInt = #line) {
+            let actual = theGroup.file(with: filePath)
+            if let expected = hasReference {
+                XCTAssertEqual(actual, expected)
+            } else {
+                XCTAssertNil(actual)
+            }
+        }
+
+        assert(
+            filePath: file1Path,
+            hasReference: fileReferenceFor[file1Path]
+        )
+        assert(
+            filePath: file2Path,
+            hasReference: fileReferenceFor[file2Path]
+        )
+        assert(
+            filePath: groupDir + subDirectoryName + ".." + fileName,
+            hasReference: fileReferenceFor[file1Path]
+        )
+        assert(
+            filePath: groupDir + "foo/.." + subDirectoryName + fileName,
+            hasReference: fileReferenceFor[file2Path]
+        )
+        assert(
+            filePath: Path(UUID().uuidString),
+            hasReference: nil
+        )
```

Comment:
This got a little involved, but the basic idea is to recreate Xcode-like scenarios:

- You have a `projectDir` which can be located anywhere on disk
- You have a `group`, which has some path relative to parents, or absolute, etc. (as expected in Xcode)
- The `group` contains files that have paths that likewise relate to the group, or absolute etc., might have messy traversals to parent directories
- Specifics of these are irrelevant, since internally we generate those normalized absolute paths